coverage-report-generator/pom.xml

```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (c) 2010-2021. Axon Framework
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~    http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```

Comment:
Is this the cleanest way to generate coverage reports?
Furthermore, this is still executed, right? Might've missed the right profile selection there, so just making sure.

Reply:
The reports are generated by the "report" goal of the standard maven lifecycle on the "post-test" (doesn't exist as a phase, it is package) and "post-integration-test" phases. Since we separate unit from the rest and in principle it is ok to run an test of `spring-autoconfigure` module which will execute code from the `extension module` it is advisable not only to generate local reports for every artifact, but also generate an aggregate report, which includes classes of all referenced maven modules (following the compile dependencies). This is the way to create an aggregate report.

As for the generation of standard reports - it takes about 1/2 second on a decent machine is in my opinion not worth to switch off or on....

Reply:
After my comment I did some digging and noted the same as you described.
I was looking for something like this some time ago, as the main Framework repo has some tests that aren't in the same module as the code they test.
Due to this, SonarCloud always complains for missed coverage, which is plain wrong.

If I understand correctly, we could solve that (here, and in Framework (or anywhere else really)) by using such a separate pom file to aggregate all the coverage reports.
I am rephrasing that correctly I think, right @zambrovski?

Reply:
@smcvb you are right. The aggregate report will help you with this. Actually you need it as soon as you have more than one module in Maven. In our case it is even more: we separated tests to Unit and Integration, so to measure the whole coverage you need the aggregate report.

Reply:
https://www.eclemma.org/jacoco/trunk/doc/report-aggregate-mojo.html